metricflow/specs/patterns/entity_link_pattern.py

```diff
@@ -0,0 +1,174 @@
+        keys_to_check = set(field_to_compare.value for field_to_compare in self.parameter_set.fields_to_compare)
+        asdict(self.parameter_set)
+        # Checks that EntityLinkPatternParameterSetField is valid wrt to the parameter set.
+        parameter_set_values = tuple(getattr(self.parameter_set, key_to_check) for key_to_check in keys_to_check)
+
+        for spec in candidate_specs:
+            spec_values = tuple(
+                (getattr(spec, key_to_check) if hasattr(spec, key_to_check) else None) for key_to_check in keys_to_check
+            )
+            if spec_values == parameter_set_values:
+                matching_specs.append(spec)
+
+        return matching_specs
+
+
+@dataclass(frozen=True)
+class DimensionPattern(EntityLinkPattern):
+    """Similar to EntityPathPattern but only matches dimensions / time dimensions.
+
+    Analogous pattern for Dimension() in the object builder naming scheme.
+    """
+
+    @override
+    def match(self, candidate_specs: Sequence[LinkableInstanceSpec]) -> Sequence[LinkableInstanceSpec]:
+        spec_set = LinkableSpecSet.from_specs(tuple(candidate_specs))
+        filtered_specs: Sequence[LinkableInstanceSpec] = spec_set.dimension_specs + spec_set.time_dimension_specs
+        return super().match(filtered_specs)
+
+    @staticmethod
+    def from_call_parameter_set(  # noqa: D
+        dimension_call_parameter_set: DimensionCallParameterSet,
+    ) -> DimensionPattern:
+        return DimensionPattern(
+            parameter_set=EntityLinkPatternParameterSet(
+                fields_to_compare=(
+                    ParameterSetField.ELEMENT_NAME,
+                    ParameterSetField.ENTITY_LINKS,
+                ),
+                element_name=dimension_call_parameter_set.dimension_reference.element_name,
+                entity_links=dimension_call_parameter_set.entity_path,
+            )
+        )
+
+
+@dataclass(frozen=True)
+class TimeDimensionPattern(EntityLinkPattern):
+    """Similar to EntityPathPattern but only matches time dimensions.
+
+    Analogous pattern for TimeDimension() in the object builder naming scheme.
+    """
+
+    @override
+    def match(self, candidate_specs: Sequence[LinkableInstanceSpec]) -> Sequence[LinkableInstanceSpec]:
+        spec_set = LinkableSpecSet.from_specs(tuple(candidate_specs))
+        return super().match(spec_set.time_dimension_specs)
+
+    @staticmethod
+    def from_call_parameter_set(  # noqa: D
+        time_dimension_call_parameter_set: TimeDimensionCallParameterSet,
+    ) -> TimeDimensionPattern:
+        fields_to_compare: List[ParameterSetField] = [
+            ParameterSetField.ELEMENT_NAME,
+            ParameterSetField.ENTITY_LINKS,
+            ParameterSetField.DATE_PART,
+        ]
+
+        if time_dimension_call_parameter_set.time_granularity is not None:
+            fields_to_compare.append(ParameterSetField.TIME_GRANULARITY)
```

Comment:
It's a little puzzling to me why one of these is using an is not None check but not the other. Is there any reason not to just compare both of these every time? Why do we care if the granularity is set for the purposes of the `match` call?

It appears the reason is so we can use None as a value for "any time granularity" here. Regardless, adding a docstring to the method explaining this instead of suppressing lint would be great.

Reply:
Yeah, that's the case to follow the behavior that's defined for how the time grain / date part are used in the object builder naming scheme. Added a comment.